### PR TITLE
client: Add context parameter and enable tracing support

### LINKED ISSUE #327: Add gRPC tracing for clients
Add a `context.Context` parameter to the client `NewAgentClient()` API and enable gRPC tracing if the specified context contains an opentracing span.

### PR DESCRIPTION
**Pull request by jodh-intel**
Add a `context.Context` parameter to the client `NewAgentClient()` API and
enable gRPC tracing if the specified context contains an opentracing
span.

Fixes #327.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Note that this change is somewhat "invasive" since it adds an opentracing dependency to the client. However, that's the direction we want to move in anyway (see https://github.com/kata-containers/kata-containers/issues/27).

Note also that the gRPC *server* the agent runs is not currently traced - that will be dealt with on #322.

**Comment by devimc** [comment]
CI is not happy

```
protocols/client/client_test.go:96:35:warning: ...
```

**Comment by jodh-intel** [comment]
Branch updated.

File: protocols/client/client.go [hunk 1]
```diff
@@ -54,14 +56,28 @@ type dialer func(string, time.Duration) (net.Conn, error)
 //   - unix://<unix socket path>
 //   - vsock://<cid>:<port>
 //   - <unix socket path>
-func NewAgentClient(sock string, enableYamux bool) (*AgentClient, error) {
+func NewAgentClient(ctx context.Context, sock string, enableYamux bool) (*AgentClient, error) {
```
Comment (devimc) [inline protocols/client/client.go]: why this change? the context is always `Background()`

Comment (jodh-intel) [reply protocols/client/client.go]: We want to be able to pass the context the runtime creates to `virtcontainers/kata_agent.go` because the runtime's context will have the trace information stored inside it so with this change, when the runtime is executed, we'll get gRPC traces for the client (runtime) side.

Comment (devimc) [reply protocols/client/client.go]: make sense, thanks
